### ISSUE TITLE
Implement torch.clamp() on sparse tensors with SparseCPU backend

### ISSUE THREAD

**Issue by samcorbettdavies** [pytorch/pytorch#82494]
### 🚀 The feature, motivation and pitch

torch.clamp() fails on sparse tensors using SparseCPU backend. Among other things, this breaks gradient clipping with sparse tensors. 

```
import torch

sparse_tensor = torch.sparse_coo_tensor([[1,2]], [1,5], (3,))
torch.clamp(sparse_tensor, -1, 1)
```

Fails with error: `Could not run 'aten::clamp' with arguments from the 'SparseCPU' backend.`

### Alternatives

It's not exactly difficult to manually clamp sparse tensors, but we shouldn't have to when a function exists.

**Comment by amjames** [comment]
This is an interesting case as we have so far explicitly avoided support for ops which will map zero->nonzero. 

Would a sparse implementation for clamp need to check the min/max values and throw if zero is not preserved?

**Comment by samcorbettdavies** [comment]
Interesting, good point. I was thinking about use cases where we want to constrain the absolute values of the entries.

> Would a sparse implementation for clamp need to check the min/max values and throw if zero is not preserved?
This makes sense to me, if we want to continue to avoid ops which will map zero->nonzero.

For what it's worth, the error message I get says that clamp is implemented in the SparseCsrCPU and SparseCsrCUDA backends.

**Comment by amjames** [comment]
> Interesting, good point. I was thinking about use cases where we want to constrain the absolute values of the entries.

That is for sure the main use case for clamp, I am just pointing out that it is *possible* to use it in a way that we might want to forbid. You could formally think of `clamp` as defining a family of operations, where the min/max values are parameters every combination of which define a particular function in the family. Some (infinite) subsets of those parameters may need to be forbidden for sparse inputs because they destroy sparsity. 

> For what it's worth, the error message I get says that clamp is implemented in the SparseCsrCPU and SparseCsrCUDA backends.

Probably a good idea to see what those implementations do.

**Comment by amjames** [comment]
This came up in a conversation with the sparse team and the general consensus is that `clamp` should throw if the min/max arguments are such that zeros will not be preserved (for sparse inputs).
